object/permission.go

```diff
@@ -107,7 +108,7 @@ func getPermission(owner string, name string) (*Permission, error) {
 	if existed {
 		return &permission, nil
 	} else {
-		return nil, nil
+		return nil, errors.New("permission not found")
```

Comment:
Doesn't exist is not an error. See all other objects. It will break a lot of things. Modify the logic more outside if needed

Reply:
so it should be return false when not found a permission?

Reply:
Do you search all occurences of such cases? The semantics now is that "if something doesn't exist, it doesn't need to be an error, maybe just doesn't exist".

![image](https://github.com/casdoor/casdoor/assets/3787410/f09adbbd-4cc2-41db-9e25-8840aefbf44f)

If you change that semantics in one place, then you should change all occurences. And changing all without adapting other parts of code like frontend will break the system, the UI.

Reply:
If you are only fixing for your encountered issue at Enforce() or BatchEnforce() API in controllers, then just move your modification to controllers, don't change in object package.

Reply:
@hsluoyz please review